contracts/PushComm/PushCommV3.sol

```diff
@@ -689,21 +689,33 @@ contract PushCommV3 is Initializable, PushCommStorageV2, IPushCommV3, PausableUp
      */
     // Should be only admin
     // Should only bridge NTT TOKENS FROM COMM TO CORE on ethereum
-    function transferFeePoolToCore(uint256 _amount) external payable onlyPushChannelAdmin {
+    function transferFeePoolToCore(CrossChainRequestTypes.CrossChainFunction functionType, uint256 _amount, uint256 gasLimit) external payable onlyPushChannelAdmin {
         if (PROTOCOL_POOL_FEE < _amount) {
             revert Errors.InsufficientFunds();
         }
-
-        bytes32 recipient = bytes32(uint256(uint160(EPNSCoreAddress)));
-
+        uint256 messageBridgeCost = quoteMsgRelayCost(WORMHOLE_RECIPIENT_CHAIN, gasLimit);
         uint256 tokenBridgeCost = quoteTokenBridgingCost();
-        if (msg.value < tokenBridgeCost) {
+
+        if (msg.value < (messageBridgeCost + tokenBridgeCost)) {
             revert Errors.InsufficientFunds();
         }
 
         PROTOCOL_POOL_FEE = PROTOCOL_POOL_FEE - _amount;
 
         PUSH_NTT.approve(address(NTT_MANAGER), _amount);
```

Comment:
Gas#4: NTT_MANAGER interface could be stored in a memory variable and then use

Reply:
why? NTT_MANAGER state is only used once. @nil

Reply:
1. https://github.com/push-protocol/push-smart-contracts/blob/796b637986b8a3aa1f5841395dec463c520186dd/contracts/PushComm/PushCommV3.sol#L659
2. https://github.com/push-protocol/push-smart-contracts/blob/796b637986b8a3aa1f5841395dec463c520186dd/contracts/PushComm/PushCommV3.sol#L660

Reply:
the one on line 4 is a function call using NTT_MANAGER. will it help using local variable in this case...if we have the gas data on this we can include this change else skip. @0xNilesh

Reply:
I tried it a few days ago in remix, and it definitely helps
I can get you results if you need